packages/astro/src/default/components/MetaTags.astro

```diff
@@ -0,0 +1,36 @@
+---
+import type { MetaTagsConfig } from '@tutorialkit/types';
+import { readFaviconFile } from '../utils/favicon';
+import { readLogoFile } from '../utils/logo';
+import { readPublicImage } from '../utils/publicImage';
+
+interface Props {
+  meta?: MetaTagsConfig;
+}
+const { meta = {} } = Astro.props;
+const site = Astro.site?.toString();
+let imageUrl;
+if (meta.image) {
+  imageUrl = readPublicImage(meta.image, true);
+  if (!imageUrl) {
+    console.warn(`Image ${meta.image} not found in "/public" folder`);
+  }
+}
+imageUrl ??= readLogoFile('logo', true);
+const faviconUrl = readFaviconFile('favicon', true);
+---
+
+<meta charset="UTF-8" />
+<meta name="viewport" content="width=device-width" />
+<meta name="generator" content={Astro.generator} />
+{meta.description ? <meta name="description" content={meta.description} /> : null}
+{meta.title ? <title>{meta.title}</title> : null}
+{faviconUrl ? <link rel="icon" type="image/svg+xml" href={faviconUrl} /> : null}
```

Comment:
This should keep using the `import.meta.BASE_URL`. Also this isn't a meta tag, like `<title>`, let's move them back to where they were.

Reply:
Yes it does use BASE_URL but wrapped within readPublicImage, so the path management is centralized in a single function